pkg/server/server.go

```diff
@@ -187,9 +187,12 @@ func (h *handler) authenticateEndpoint(w http.ResponseWriter, req *http.Request)
 	var username string
 	var groups []string
 	if ec2RoleMapping, exists := h.lowercaseEC2RoleMap[arnLower]; exists {
+		// username must be a DNS-1123 hostname matches the regex "[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+		specifiedRole := strings.Replace(identity.SessionName, "@", "-", -1)
 		username = ec2RoleMapping.UsernameFormat
 		username = strings.Replace(username, "{{AccountID}}", identity.AccountID, -1)
 		username = strings.Replace(username, "{{InstanceID}}", identity.SessionName, -1)
+		username = strings.Replace(username, "{{CallerSpecifiedRoleName}}", specifiedRole, -1)
```

Comment:
@mattmoyer Thanks for the review! I'm ok with merging the `InstanceID` and `CallerSpecifiedRoleName` into just `SessionName` as you've suggested.
However then, would you mind if I blindly replace every occurrence of `@` in a session name to `-` like I did in the line 191? (Would be ok as it won't affect instance ids - just confirming assuming it would look a bit dirty?)